### PR TITLE
fix the bug of mixed precision training for unconditional generation

### PR DESCRIPTION
**Pull request by LYMDLUT**
Add accelerator.autocast(): to fix the bug of mixed precision training for "examples/unconditional_image_generation/train_unconditional.py"

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Hey @LYMDLUT,

Can you share more about what "bug" is solved by this. I'm not sure we want to apply autocast here this way . cc @williamberman @sayakpaul as well

**Comment by LYMDLUT** [comment]
Exception has occurred: RuntimeError
Found dtype Half but expected Float
  File "/data4/lym/final_code/train/train_unconditional_visual_from_pretrained.py", line 609, in main
    accelerator.backward(loss)
  File "/data4/lym/final_code/train/train_unconditional_visual_from_pretrained.py", line 741, in <module>
    main(args)
RuntimeError: Found dtype Half but expected Float

**Comment by LYMDLUT** [comment]
> Hey @LYMDLUT,
> 
> Can you share more about what "bug" is solved by this. I'm not sure we want to apply autocast here this way . cc @williamberman @sayakpaul as well

/.cursor-server/extensions/ms-python.python-2023.14.0/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher 33419 -- /data2/lyc/anaconda3/envs/pt2/bin/accelerate launch --num_processes=1 train_unconditional_visual_from_pretrained.py --dataset_name=cifar10 --resolution=32 --center_crop --random_flip --output_dir=ddpm-cifar10-100e-clean-fintune-pretrained --train_batch_size=128 --num_epochs=100 --checkpointing_steps=2000 --use_ema --model_config_name_or_path ./ddpm_ema_cifar10 --mixed_precision fp16 
09/16/2023 02:29:22 - INFO - __main__ - Distributed environment: MULTI_GPU  Backend: nccl
Num processes: 1
Process index: 0
Local process index: 0
Device: cuda:0

Mixed precision type: fp16

{'upsample_type', 'downsample_type', 'dropout'} was not found in config. Values will be initialized to default values.
.py:23: FutureWarning: transformers.deepspeed module is deprecated and will be removed in a future version. Please import deepspeed modules directly from 


Using the latest cached version of the module from /home/liuyuchen/.cache/huggingface/modules/datasets_modules/datasets/cifar10/447d6ec4733dddd1ce3bb577c7166b986eaa4c538dcd9e805ba61f35674a9de4 (last modified on Wed Jul 19 22:06:30 2023) since it couldn't be found locally at cifar10., or remotely on the Hugging Face Hub.
09/16/2023 02:31:03 - WARNING - datasets.load - Using the latest cached version of the module from /home/liuyuchen/.cache/huggingface/modules/datasets_modules/datasets/cifar10/447d6ec4733dddd1ce3bb577c7166b986eaa4c538dcd9e805ba61f35674a9de4 (last modified on Wed Jul 19 22:06:30 2023) since it couldn't be found locally at cifar10., or remotely on the Hugging Face Hub.
09/16/2023 02:31:03 - INFO - __main__ - Dataset size: 50000
09/16/2023 02:31:04 - INFO - __main__ - ***** Running training *****
09/16/2023 02:31:04 - INFO - __main__ -   Num examples = 50000
09/16/2023 02:31:04 - INFO - __main__ -   Num Epochs = 100
09/16/2023 02:31:04 - INFO - __main__ -   Instantaneous batch size per device = 128
09/16/2023 02:31:04 - INFO - __main__ -   Total train batch size (w. parallel, distributed & accumulation) = 256
09/16/2023 02:31:04 - INFO - __main__ -   Gradient Accumulation steps = 2
09/16/2023 02:31:04 - INFO - __main__ -   Total optimization steps = 19600
Epoch 0:   0%|                                                                                                                    | 0/196 [00:00<?, ?it/s]

**Comment by LYMDLUT** [comment]
@patrickvonplaten

File: examples/unconditional_image_generation/train_unconditional.py [hunk 1]
```diff
@@ -575,22 +575,24 @@ def transform_images(examples):
...
-                        noise_scheduler.alphas_cumprod, timesteps, (clean_images.shape[0], 1, 1, 1)
-                    )
-                    snr_weights = alpha_t / (1 - alpha_t)
-                    loss = snr_weights * F.mse_loss(
-                        model_output, clean_images, reduction="none"
-                    )  # use SNR weighting from distillation paper
-                    loss = loss.mean()
-                else:
-                    raise ValueError(f"Unsupported prediction type: {args.prediction_type}")
+                # Support mixed precision training
```
Comment (sayakpaul) [inline examples/unconditional_image_generation/train_unconditional.py:578]: We don't use autocast in the examples like this. Rest of the examples don't seem to suffer from this issue. So, it'd be nice to know where does it stem from.

Comment (LYMDLUT) [reply examples/unconditional_image_generation/train_unconditional.py:578]: Could you provide an example of how to use autocast in accelerate?

**Pull request closed without merging**